components/style/properties/longhand/svg.mako.rs

```diff
@@ -49,3 +49,58 @@ ${helpers.predefined_type(
 // https://www.w3.org/TR/css-masking-1/
 ${helpers.single_keyword("mask-type", "luminance alpha",
                          products="gecko", animatable=False)}
+
+<%helpers:longhand name="clip-path" animatable="False" products="gecko">
+    use cssparser::ToCss;
+    use std::fmt;
+    use values::LocalToCss;
+    use values::NoViewportPercentage;
+    use values::specified::basic_shape::{ShapeSource, GeometryBox};
+
+    // NB: `top` and `left` are 0 if `auto` per CSS 2.1 11.1.2.
+
+    pub mod computed_value {
+        use app_units::Au;
+        use values::computed::basic_shape::{ShapeSource, GeometryBox};
+
+        #[derive(Debug, Clone, PartialEq)]
+        #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+        pub struct T(pub ShapeSource<GeometryBox>);
```

Comment:
I'm not yet sure. I plan to file an E-less-easy bug for someone to investigate this.

It seems to work in some cases but not in others IIRC